hedera-mirror-rest/middleware/requestNormalizer.js

```diff
@@ -0,0 +1,81 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getOpenApiMap} from './openapiHandler.js';
+import {filterKeys} from '../constants.js';
+
+const openApiMap = getOpenApiMap();
+
+// Multiple values of these params can be collapsed to the last value
+const COLLAPSABLE_PARAMS = [filterKeys.BALANCE, filterKeys.NONCE, filterKeys.SCHEDULED];
+
+/**
+ * Do not sort these query parameters as the results of the sql query changes based on their order
+ *
+ * Some examples from the spec tests:
+ *   /api/v1/contracts/results?block.number=11&block.number=10 sorts the results differently than ?block.number=10&block.number=11
+ *
+ * From historical-custom-fees.json:
+ *   /api/v1/tokens/1135?timestamp=lt:1234567899.999999000&timestamp=1234567899.999999001 returns a result whereas
+ *   ?timestamp=1234567899.999999001&timestamp=lt:1234567899.999999000 returns a 404
+ */
+const NON_SORTED_PARAMS = COLLAPSABLE_PARAMS.concat([
+  filterKeys.BLOCK_HASH,
+  filterKeys.BLOCK_NUMBER,
+  filterKeys.TIMESTAMP,
+]);
+
+/**
+ * Normalizes a request by adding any missing default values and sorting any array query parameters.
+ *
+ * It is expected that this is called after any error handling for the request
+ *
+ * @param openApiRoute {string}
+ * @param path {string}
+ * @param query request query object
+ * @returns {string}
+ */
+const normalizeRequestQueryParams = (openApiRoute, path, query) => {
+  const openApiParameters = openApiMap.get(openApiRoute);
```

Comment:
Is it possible the route is ever not found? Maybe add a check and a test just in case.
```suggestion
  const openApiParameters = openApiMap.get(openApiRoute) || {};
```

Reply:
I tried this, or something similar: ` const openApiParameters = openApiMap.get(openApiRoute) || []` and added a test case for a request that has no matching openApi parameters. I found that if that request had query parameters that the current iteration of the method would remove them. So instead I added a check for the open api parameters, and if missing, then the normalized query will be generated with querystring.